app/uk/gov/hmrc/tariffclassificationfrontend/controllers/LiabilityController.scala

```diff
@@ -35,23 +39,103 @@ import scala.concurrent.Future.successful
 @Singleton
 class LiabilityController @Inject()(verify: RequestActions,
                                     val messagesApi: MessagesApi,
+                                    casesService: CasesService,
                                     implicit val appConfig: AppConfig) extends FrontendController with I18nSupport {
 
   private lazy val menuTitle = LIABILITY
 
+  private lazy val form = LiabilityFormData.form
+
   def liabilityDetails(reference: String): Action[AnyContent] = (verify.authenticated andThen verify.casePermissions(reference)).async { implicit request =>
-    getCaseAndRenderView(
-      menuTitle,
+    getCaseAndRenderView(menuTitle,
       c => {
         successful(views.html.partials.liability_details(c, tabIndexFor(LIABILITY)))
       }
     )
   }
 
+  def editLiabilityDetails(reference: String): Action[AnyContent] = (verify.authenticated andThen verify.casePermissions(reference)).async { implicit request =>
+    request.`case` match {
+      case c: Case =>
+        successful(
+          Ok(
+            liability_details_edit(c, toLiabilityForm(c.application.asLiabilityOrder), Some(tabIndexFor(LIABILITY)))
+          )
+        )
+    }
+  }
+
+  private def toLiabilityForm(l: LiabilityOrder): Form[LiabilityFormData] = {
```

Comment:
Can we try and do this with the form mapping. See `DecisionFormData` with the Liability form